pkg/apis/resource/types.go

```diff
@@ -907,3 +919,77 @@ type ResourceClaimTemplateList struct {
 	// Items is the list of resource claim templates.
 	Items []ResourceClaimTemplate
 }
+
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+	// Driver specifies the name of the DRA driver whose kubelet
+	// plugin should be invoked to process the allocation once the claim is
+	// needed on a node.
+	//
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	//
+	// +required
+	Driver string
+
+	// This name together with the driver name and the device name field
+	// identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+	//
+	// Must not be longer than 253 characters and may contain one or more
+	// DNS sub-domains separated by slashes.
+	//
+	// +required
+	Pool string
+
+	// Device references one device instance via its name in the driver's
+	// resource pool. It must be a DNS label.
+	//
+	// +required
+	Device string
+
+	// Conditions contains the latest observation of the device's state.
+	// If the device has been configured according to the class and claim
+	// config references, the `Ready` condition should be True.
+	//
+	// +optional
+	// +listType=atomic
+	Conditions []metav1.Condition
+
+	// Data contains arbitrary driver-specific data.
+	//
+	// +optional
+	Data *runtime.RawExtension
+
+	// NetworkData contains network-related information specific to the device.
+	//
+	// +optional
+	NetworkData *NetworkDeviceData
+}
+
+// NetworkDeviceData provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
+type NetworkDeviceData struct {
+	// InterfaceName specifies the name of the network interface associated with
+	// the allocated device. This might be the name of a physical or virtual
+	// network interface.
+	//
+	// +optional
+	InterfaceName *string
```

Comment:
@thockin another question about pointer vs empty, an interfacename with an empty string is valid, so in this case having a pointer sseems worse than using the empty value, since we need to validate `if InterfaceName != nil then; if  len(interfaceName) >0 `

Reply:
@liggitt we talked offline about this the other day? for string values that the empty value is not meaningul I think is better to not use pointers, since it does not matter is nil or empty, you will ignore the field anyway ... WDYT?

Reply:
discussed in api machinery channel https://kubernetes.slack.com/archives/CJUQN3E4T/p1730109344106609 @LionelJouin can you remove the pointer, a valid interface name can never be empty so I don't think we need to disambiguate here. ... same for HWaddress https://github.com/kubernetes/kubernetes/pull/128240/files#r1809451042